lib/active_fedora/with_metadata/metadata_node.rb

```diff
@@ -70,6 +70,10 @@ def changes_for_update
           ChangeSet.new(self, self, changed_attributes.keys).changes
         end
 
+        def server_managed_properties
+          @server_managed_properties ||= properties.select { |k,v| v[:server_managed] }.symbolize_keys.keys
```

Comment:
this is minor, but i think this will have notably fewer allocations:

```suggestion
          @server_managed_properties ||= properties.select { |k,v| v[:server_managed] }.keys.map(&:to_sym)
```